src/plugins/kibana_utils/public/state_management/url/kbn_url_tracker.ts

```diff
@@ -148,9 +157,13 @@ export function createKbnUrlTracker({
     unsubscribe();
     // track current hash when within app
     unsubscribeURLHistory = historyInstance.listen(location => {
-      const currentAppName = defaultSubUrl.slice(2); // cut hash and slash
+      const currentAppName = defaultSubUrl.slice(2); // cut hash and slash symbols
       const targetAppName = location.pathname.split('/')[1];
-      if (currentAppName === targetAppName || currentAppName.startsWith(targetAppName)) {
+
+      if (
```

Comment:
This implementation only allows the caller of kbn url tracker to overwrite the behavior if the target app name is not matching the current app name. What about going one step further and allowing the user to also overwrite this behavior?

A straight-forward implementation would be to add a default implementation for the `isUrlBelongsToApp`/`shouldTrackUrlUpdate` parameter which only does  `currentAppName === targetAppName` - if the caller passes in another implementation, it is overwritten.

Reply:
Done.